src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -351,6 +351,17 @@ ZooKeeper::ZooKeeper(
     }
 
     connect(nodes, args.connection_timeout_ms * 1000);
+    if (args.session_lifetime.has_value())
+    {
+        std::uniform_int_distribution<UInt64> distribution
+        {
+            args.session_lifetime->min_sec,
+            args.session_lifetime->max_sec,
+        };
+        UInt64 client_session_duration_sec = distribution(thread_local_rng);
+        client_session_deadline = clock::now() + std::chrono::seconds(client_session_duration_sec);
```

Comment:
Will this work with system clock changes?

Reply:
Thank you for taking a look!
I'm not 100% sure. I'm following an example in src/Dictionaries/SSDCacheDictionaryStorage.h:1366 , does that account for system clock changes?

Reply:
```
using clock = std::chrono::steady_clock;
```
[Class std::chrono::steady_clock represents a monotonic clock.](https://en.cppreference.com/w/cpp/chrono/steady_clock)

So it's okay.